app/reducers/galleryPictures.js

```diff
@@ -13,17 +31,72 @@ export type GalleryPictureEntity = {
   description: string,
   text: string,
   active: boolean,
-  comments: Array<number>
+  comments: Array<number>,
+  file: string,
+  thumbnail: string,
+  rawFile: string
 };
 
 function mutateGalleryPicture(state: any, action: any) {
+  const { uploadStatus = initialUploadStatus } = state;
   switch (action.type) {
     case GalleryPicture.DELETE.SUCCESS: {
       return {
         ...state,
         items: state.items.filter(id => id !== action.meta.id)
       };
     }
+    case Gallery.UPLOAD.BEGIN: {
+      const imageCount = uploadStatus.imageCount + action.meta.imageCount;
```

Comment:
This imagecount includes pictures from previous uploads, even if they were in different albums. The only way to reset it is to refresh the page. You could argue that's a feature I suppose, but I would argue each upload is separate and should have its own count. From a user perspective, a redux state "session" is a pretty weird way to decide when to reset the counter.

Reply:
Yeah.

Kinda copied the functionality from google photos. If someone uploads to two galleries at the same time, what should we do then? :thinking: Or if a user does multiple uploads to the same gallery, at the same time?

Reply:
Can't we just disallow that? "Vennligst fullfør opplastingen din først"?